tests/integration/test_publish_flow.py

```diff
@@ -2,31 +2,36 @@
 # Copyright 2021 Ocean Protocol Foundation
 # SPDX-License-Identifier: Apache-2.0
 #
+from ocean_lib.agreements.file_objects import FilesTypeFactory
+from ocean_lib.agreements.service_types import ServiceTypesV4
+from ocean_lib.data_provider.data_service_provider import DataServiceProvider
 from ocean_lib.models.v4.erc721_factory import ERC721FactoryContract
-from ocean_lib.models.v4.erc721_token import ERC721Token
 from ocean_lib.models.v4.models_structures import ErcCreateData
+from ocean_lib.ocean.v4.ocean_assets import OceanAssetV4
+from ocean_lib.services.v4.service import V4Service
+from ocean_lib.utils.utilities import get_timestamp
 from ocean_lib.web3_internal.constants import ZERO_ADDRESS
-from tests.resources.helper_functions import get_address_of_type
+from tests.resources.ddo_helpers import build_credentials_dict
+from tests.resources.helper_functions import get_address_of_type, deploy_erc721_erc20
 
 
-# TODO: WIP - draft publish flow
 def test_publish_flow(web3, config, publisher_wallet):
-
+    """Tests publish flow in different situations."""
     erc721_factory_address = get_address_of_type(
         config, ERC721FactoryContract.CONTRACT_NAME
     )
     erc721_factory = ERC721FactoryContract(web3, erc721_factory_address)
 
     # Publisher deploys NFT contract
     tx = erc721_factory.deploy_erc721_contract(
```

Comment:
I think this call was was changed, so we need to rebase with v4main after merging Pablo's new artifacts work.